test/suite/test_prepare02.py

```diff
@@ -0,0 +1,123 @@
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.alter("table:mytable",
+                "access_pattern_hint=random"), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.create("table:mytable1",
+                "key_format=S,value_format=S"), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.compact("table:mytable"), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.drop("table:mytable", None), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.join(cursor, cursor,
+                "compare=gt,count=10"), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.log_flush("sync=on"), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.log_printf("Printing to log file"), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.rebalance("table:mytable", None), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.rename("table:mytable", "table:mynewtable",
+                None), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda:self.session.reset(), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.salvage("table:mytable", None), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.truncate("table:mytable",
+                None, None, None), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.upgrade("table:mytable", None), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.verify("table:mytable", None), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda:self.session.begin_transaction(), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda:self.session.prepare_transaction(), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.timestamp_transaction(
+                "commit_timestamp=2a"), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda:self.session.checkpoint(), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.snapshot("name=test"), msg)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda:self.session.transaction_sync(), msg)
+        self.session.rollback_transaction()
+
+        # Commit after prepare is permitted
```

Comment:
Minor nit - add a period at the end of the sentences in the next three comments.